apps/total-typescript/src/components/team/index.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import SelfRedeemButton from './self-redeem-button'
+import CopyInviteLink from './copy-invite-link'
+import Link from 'next/link'
+
+type InviteTeamProps = {
+  purchase: {
+    merchantChargeId: string | null
+    bulkCoupon: {id: string; maxUses: number; usedCount: number} | null
+    product: {id: string; name: string}
+  }
+  existingPurchase: {
+    id: string
+    product: {id: string; name: string}
+  }
+  session: any
+  setPersonalPurchase: (props: any) => void
+}
+
+const InviteTeam: React.FC<React.PropsWithChildren<InviteTeamProps>> = ({
+  purchase,
+  existingPurchase,
+  session,
+  setPersonalPurchase,
+}) => {
+  const redemptionsLeft =
+    purchase.bulkCoupon &&
+    purchase.bulkCoupon.maxUses > purchase.bulkCoupon.usedCount
+  const numberOfRedemptionsLeft =
+    purchase?.bulkCoupon &&
+    purchase?.bulkCoupon.maxUses - purchase.bulkCoupon.usedCount
+
+  const [canRedeem, setCanRedeem] = React.useState(
+    Boolean(redemptionsLeft && !existingPurchase),
+  )
+  const userEmail = session?.user?.email
+  const bulkCouponId = purchase?.bulkCoupon?.id
```

Comment:
*not this PR*: I'd be interested in finding a `packages` place for this kind of logic, since it is now spread across a couple apps, I wonder if it some of this could be _computed values_ provided through the DB SDK when getting bulk coupon info.

Reply:
i'm disliking the "database" package naming, but that is probably the spot for this kind of thing.